File: 3517 - Smallest Palindromic Rearrangement I/Method 1/solution.js
/**
 * @param {string} s
 * @return {string}
 */
var smallestPalindrome = function(s) {
    let n = s.length;
    if(n<=3) return s;
    let str = s.substring(0,n/2);
    let sorted = str.split("").sort().join("");
    let reversed = sorted.split("").reverse().join("");
    let newStr;
    if(n%2==1) {
        newStr = sorted + s[Math.floor(n/2)] + reversed;
    } else {
        newStr = sorted + reversed;
    }
    return newStr;
};

File: 3517 - Smallest Palindromic Rearrangement I/Method 1/solution.java
class Solution {
    public String smallestPalindrome(String s) {
        int n = s.length();
        if(n<=3) return s;
        String str = s.substring(0,n/2);
        char[] chars = str.toCharArray();
        Arrays.sort(chars);
        StringBuilder sb = new StringBuilder();
        for(int i = 0; i<chars.length; i++) {
            sb.append(chars[i]);
        }
        if(n%2==1) {
            str = sb.toString() + s.charAt(n/2) + sb.reverse().toString();
        } else {
            str = sb.toString() + sb.reverse().toString();
        }
        return str;
    }
}